### ISSUE TITLE
Allow the factory passed to IStreamClientEndpoint.connect to return None from buildProtocol

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6976]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6976|
|Type|enhancement|
|Created|2014-02-17 19:43:10Z|

This is parallel to the support for this behavior in `reactor.connectXXX` introduced in [#4598](https://github.com/twisted/twisted/issues/4598).

<details><summary>Searchable metadata</summary>

```
trac-id__6976 6976
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1392666190000000 1392666190000000
changetime__1404436172997188 1404436172997188
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

I fixed this already in the [#4598](https://github.com/twisted/twisted/issues/4598) branch.  I'll resolve at the same time as I merge that change.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `closed`|
|-|-|

(In [42850]) Merge connector-buildProtocol-710-2

Author: exarkun
Reviewer: rwall
Fixes: [#4598](https://github.com/twisted/twisted/issues/4598)
Fixes: [#6976](https://github.com/twisted/twisted/issues/6976)

Explicitly handle the case where `None` is returned from `buildProtocol`
when using any of the implementations of either `IReactorTCP.connectTCP`
or `IStreamClientEndpoint.connect` in Twisted.
